File: app/src/main/java/com/elegion/android/util/GsonUtils.java
package com.elegion.android.util;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

/**
 * @author Mike
 */
final public class GsonUtils {
    private static Gson sRequestGson;

    private static Gson sGson;

    private GsonUtils() {
    }

    public static Gson gson() {
        if (sGson == null) {
            sGson = new Gson();
        }
        return sGson;
    }

    public static Gson requestGson() {
        if (sRequestGson == null) {
            sRequestGson = new GsonBuilder()
                    .excludeFieldsWithoutExposeAnnotation()
                    .setDateFormat("yyyy-MM-dd'T'HH:mm:ss'.'SSS'Z'")
                    .create();
        }
        return sRequestGson;
    }

}

File: app/src/main/kotlin/com/elegion/android/template/util/GsonUtils.kt
package com.elegion.android.template.util

import com.google.gson.Gson
import com.google.gson.GsonBuilder

object GsonUtils {
    private lateinit var sRequestGson: Gson
    private lateinit var sGson: Gson

    @JvmStatic
    fun gson(): Gson {
        if (!::sGson.isInitialized) {
            sGson = Gson()
        }
        return sGson
    }

    @JvmStatic
    fun requestGson(): Gson {
        if (!::sRequestGson.isInitialized) {
            sRequestGson = GsonBuilder()
                .setDateFormat("yyyy-MM-dd'T'HH:mm:ss'.'SSS'Z'")
                .create()
        }
        return sRequestGson
    }
}